pkg/jmxfetch/jmxfetch.go

```diff
@@ -226,14 +227,35 @@ func (j *JMXFetch) Start() error {
 	return j.cmd.Start()
 }
 
-// Kill kills the JMXFetch process
-func (j *JMXFetch) Kill() error {
```

Comment:
I think we might want to keep around `Kill()` and `Stop()` and enforce the grace period in `stopRunner()` and perhaps rely on `(*ProcessState).Exited()` in there to check if the process came down cleanly. My main concern with calling `(*Cmd).Wait()` is that I'm not sure if the behavior is defined if we're waiting on `Wait()` from two different go routines - we'd need to check that if you want to use wait.

Reply:
What bother me a little is that we need os specific logic and I would like to keep that hidden from outside of this file. Tell me what you think.
From my testing `(*Cmd).Wait()` seems to work fine with two goroutines. 
How would you use `(*ProcessState).Exited()` ? Do something like checking it's value every N ms ?

Reply:
`(*Cmd).Process.Kill()` is portable. So my suggestion was to call `(*JMXFetch).Stop()` from `stopRunner()`, giving it a grace period there, and if the process wasn't down, calling `(*JMXFetch).Kill()` (that would call `(*Cmd).Process.Kill()`). If that makes sense. I meant something like this in the `stopRunner` logic: 

```go
r.jmxfetch.Stop()
stopped := make(chan struct{})

go func() {
  j.jmxfetch.Wait()  // so we think this is safe
  stop(closed)
}

select { 
  case <- stopped:
     return nil
  case <- time.After(gracePeriod * time.Millisecond):
     return j.jmxfetch.Kill()
}
```

Reply:
Understood, that's clearly the cleanest option on *nix, but I think we have an issue on windows. As you know we use an "Exit File" and jmxfetch checks for it every run (15s IIRC) : 
https://github.com/DataDog/jmxfetch/blob/master/src/main/java/org/datadog/jmxfetch/App.java#L278-L281
This would result in it getting killed most of the time.

Reply:
Ah, it's true that the cadence on windows is 15s for that.... I thought went you spoke about OS you were referring to signals and portability. The 15s interval checking for the file to exit or not is quite shitty to be honest. We should change that eventually. 

OK, so then your approach is the best option for now. Sorry about the confusion.